lib/migration-chunks/validation/field.js

```diff
@@ -166,17 +210,18 @@ module.exports = function (chunks, contentTypes = []) {
 
   const recentlyRemoved = {};
   const recentlyMoved = {};
+  const changedFieldIds = {};
 
   for (const chunk of chunks) {
     const contentTypeId = chunk[0].payload.contentTypeId;
     const contentTypeExists = Boolean(contentTypes.find((ct) => ct.sys.id === contentTypeId));
     const fieldSet = contentTypeFields[contentTypeId] || new Set();
     const fieldRemovals = recentlyRemoved[contentTypeId] || new Set();
     const fieldMovements = recentlyMoved[contentTypeId] || new Set();
+    const fieldIdChanges = changedFieldIds[contentTypeId] || {};
```

Comment:
Let's use a `new Map()` here, so we call `.delete(key)` methods everywhere instead of `delete foo[key]`  expression.
Same for `.add(key)` instead of `foo[key] = ...`